test/javascripts/acceptance/search-test.js.es6

```diff
@@ -73,3 +73,53 @@ QUnit.test("Search with context", assert => {
     assert.ok(!$('.search-context input[type=checkbox]').is(":checked"));
   });
 });
+
+QUnit.test("in:likes, in:private, and in:seen filters are hidden to anonymous users", assert => {
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.notOk(exists('.search-advanced-options .in-likes'));
+    assert.notOk(exists('.search-advanced-options .in-private'));
+    assert.notOk(exists('.search-advanced-options .in-seen'));
+  });
+});
+
+QUnit.test("in:likes, in:private, and in:seen filters are available to logged in users", assert => {
+  logIn();
+  Discourse.reset();
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.ok(exists('.search-advanced-options .in-likes'));
+    assert.ok(exists('.search-advanced-options .in-private'));
+    assert.ok(exists('.search-advanced-options .in-seen'));
+  });
+});
+
+QUnit.test(`"I've not read", "I posted in", "I'm watching", "I'm tracking",
```

Comment:
I would just combine these tests with the one above.

Probably just two description for the tests:

`Quni.test("right filters are shown to anonymous users")` and `Quni.test("right filters are shown to logged in users")`